File: patterns/java/src/refactoring_guru/patterns/builder/example/builders/Builder.java
package refactoring_guru.patterns.builder.example.builders;

import refactoring_guru.patterns.builder.example.cars.Type;
import refactoring_guru.patterns.builder.example.components.*;

/**
 * EN: Builder interface defines all possible ways to configure a product.
 * 
 * RU: Интерфейс Строителя объявляет все возможные этапы и шаги
 * конфигурации продукта.
 */
public interface Builder {
    public void setType(Type type);
    public void setSeats(int seats);
    public void setEngine (Engine engine);
    public void setTransmission(Transmission transmission);
    public void setTripComputer(TripComputer tripComputer);
    public void setGPSNavigator(GPSNavigator gpsNavigator);
}

File: patterns/cs/src/RefactoringGuru/Builder/Example/Builders/Builder.cs
using RefactoringGuru.Builder.Example.Components;

namespace RefactoringGuru.Builder.Example.Builders
{
    /**
     * EN: Builder interface defines all possible ways to configure a product.
     * 
     * RU: Интерфейс Строителя объявляет все возможные этапы и шаги
     * конфигурации продукта.
     */
    public interface Builder
    {
        void SetType(Cars.Type type);
        void SetSeats(int seats);
        void SetEngine(Engine engine);
        void SetTransmission(Transmission transmission);
        void SetTripComputer(TripComputer tripComputer);
        void SetGpsNavigator(GPSNavigator gpsNavigator);
    }
}
